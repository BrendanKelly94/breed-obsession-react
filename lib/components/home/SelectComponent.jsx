import React from 'react';
const SelectComponent = (props) => {

  const { items , callback} = props;

  if(items){
    return (
      <select onChange = {callback}>
        {items.map((item) => {
          return <option key = {item} value = {item}> {item} </option>;
        })}
      </select>
    );
  }else{
    return <div> Loading... </div>;
  }
};
export default SelectComponent;
