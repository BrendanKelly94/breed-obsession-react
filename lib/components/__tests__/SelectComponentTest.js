import React from 'react';
import SelectComponent from '../home/SelectComponent.jsx';

import renderer from 'react-test-renderer';

describe('SelectComponent', () => {

  const testProps = {
    items: [],
    callback: jest.fn(() => ({})),
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <SelectComponent
        {...testProps}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();

  });
});
