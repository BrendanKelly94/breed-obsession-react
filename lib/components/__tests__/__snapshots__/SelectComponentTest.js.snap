// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectComponent renders correctly 1`] = `
<select
  onChange={[MockFunction]}
/>
`;
